tests/Teams/ComposeMessagingExtensionAuth/Bots/ComposeMessagingExtensionAuthBot.cs

```diff
@@ -0,0 +1,187 @@
+using System.Threading.Tasks;
+using AdaptiveCards;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Teams;
+using Microsoft.Bot.Schema;
+using Microsoft.Bot.Schema.Teams;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.BotBuilderSamples.Bots
+{
+    /*
+     * This Bot requires an Azure Bot Service OAuth connection name in appsettings.json
+     * see: https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-authentication
+     *
+    * Clicking this bot's Task Menu will retrieve the login dialog, if the user is not already signed in.
+    */
+    public class ComposeMessagingExtensionAuthBot : TeamsActivityHandler
+    {
+        readonly string _connectionName;
+
+        public ComposeMessagingExtensionAuthBot(IConfiguration configuration)
+        {
+            _connectionName = configuration["ConnectionName"];
+        }
+
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if (!string.IsNullOrEmpty(turnContext.Activity.Text))
+            {
+                // Hack around weird behavior of RemoveRecipientMention (it alters the activity.Text)
+                string originalText = turnContext.Activity.Text;
+                turnContext.Activity.RemoveRecipientMention();
+                string text = turnContext.Activity.Text.Replace(" ", string.Empty).ToUpper();
+                turnContext.Activity.Text = originalText;
+
+                if (text.Equals("LOGOUT") || text.Equals("SIGNOUT"))
+                {
+                    await (turnContext.Adapter as IUserTokenProvider).SignOutUserAsync(turnContext, _connectionName, turnContext.Activity.From.Id, cancellationToken).ConfigureAwait(false);
+
+                    await turnContext.SendActivityAsync(MessageFactory.Text($"Signed Out: {turnContext.Activity.From.Name}"), cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    await turnContext.SendActivityAsync(MessageFactory.Text($"echo: {turnContext.Activity.Text}"), cancellationToken).ConfigureAwait(false);
+                }
+            }
+        }
+
+        protected override async Task<MessagingExtensionActionResponse> OnTeamsMessagingExtensionFetchTaskAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionQuery query, CancellationToken cancellationToken)
+        {
+            var reply = MessageFactory.Text("OnTeamsMessagingExtensionFetchTaskAsync MessagingExtensionQuery: " + JsonConvert.SerializeObject(query));
+            await turnContext.SendActivityAsync(reply, cancellationToken).ConfigureAwait(false);
+
+            var tokenStatus = await (turnContext.Adapter as IUserTokenProvider).GetTokenStatusAsync(turnContext, turnContext.Activity.From.Id, _connectionName, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
nit: we don't usually put .ConfigureAwait(false)

Reply:
Removed .ConfigureAwaits